front/temporal/labs/utils/gong.ts

```diff
@@ -0,0 +1,195 @@
+
+export async function retrieveGongTranscripts(
+  transcriptsConfiguration: LabsTranscriptsConfigurationResource,
+  localLogger: Logger
+): Promise<string[]> {
+  // Retrieve transcripts from Gong from the last 24h
+  const fromDateTime = new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString();
+
+  if (!transcriptsConfiguration.connectionId) {
+    localLogger.error(
+      {},
+      "[retrieveGongTranscripts] No connectionId found. Skipping."
+    );
+    return [];
+  }
+
+  const gongAccessToken = await getAccessTokenFromNango(
+    "gong-dev",
+    transcriptsConfiguration.connectionId
+  );
+
+  const newTranscripts = await fetch(
+    `https://api.gong.io/v2/calls?fromDateTime=${fromDateTime}`,
+    {
+      headers: {
+        Authorization: `Bearer ${gongAccessToken}`,
+        "Content-Type": "application/json",
+      },
+    }
+  );
+
+  if (!newTranscripts.ok) {
+    localLogger.error(
+      {},
+      "[retrieveNewTranscripts] Error fetching new transcripts from Gong. Skipping."
+    );
+    return [];
+  }
+
+  const newTranscriptsData = await newTranscripts.json();
+
+  if (!newTranscriptsData || newTranscriptsData.length === 0) {
+    localLogger.info(
+      {},
+      "[retrieveNewTranscripts] No new transcripts found from Gong."
+    );
+    return [];
+  }
+
+  const fileIdsToProcess = [];
+
+  for (const call of newTranscriptsData.calls) {
+    const { id: fileId } = call;
+    if (!fileId) {
+      localLogger.error(
+        {},
+        "[retrieveNewTranscripts] call does not have an id. Skipping."
+      );
+      continue;
+    }
+
+    const history = await transcriptsConfiguration.fetchHistoryForFileId(
+      fileId
+    );
+    if (history) {
+      localLogger.info(
+        { fileId },
+        "[retrieveNewTranscripts] call already processed. Skipping."
+      );
+      continue;
+    }
+
+    fileIdsToProcess.push(fileId);
+  }
+
+  return fileIdsToProcess;
+}
+
+export async function retrieveGongTranscriptContent(
+  transcriptsConfiguration: LabsTranscriptsConfigurationResource,
+  fileId: string,
+  localLogger: Logger
+): Promise<{ transcriptTitle: string; transcriptContent: string }> {
+  type GongParticipant = {
+    speakerId: string;
+    name: string;
+    [key: string]: any;
+  };
+
+  const gongAccessToken = await getAccessTokenFromNango(
+    "gong-dev",
+    transcriptsConfiguration.connectionId
+  );
+
+  const call = await fetch(`https://api.gong.io/v2/calls/extensive`, {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${gongAccessToken}`,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      filter: {
+        callIds: [fileId],
+      },
+    }),
+  });
+
+  if (!call.ok) {
+    localLogger.error(
+      {},
+      "[processTranscriptActivity] Error fetching call from Gong. Skipping."
+    );
+    throw new Error("Error fetching call from Gong. Skipping.");
+  }
+
+  const callData = (await call.json()).calls[0];
```

Comment:
Can we add typing here just to describe the expected type and have the rest of this function be a little more type safe?
Example:
```
const callData : { participans: {...}[] }
```